src/tests/JIT/opt/Devirtualization/GitHub_51918.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+
+public interface IGeneric<T>
+{
+    int InterfaceMethod();
+}
+
+public interface IMakeClassMethodSealedVirtual
+{
+    int ClassMethod();
+}
+
+public class GenericClass<A,B> : IMakeClassMethodSealedVirtual
+{
+    public static int _sv1;
+    public static int _sv2;
+
+    public int _v1;
+    public int _v2;
+
+    public GenericClass()
```

Comment:
I can find the original issue but it would be extremely helpful if you could add at least a three-line comment describing the way the test is supposed to work and how it detects bad behavior - in my recent work on fixing the various regression tests with Crossgen2 I've always had to spend quite a bit of time poring over the particular test to understand what it's supposed to be doing especially when it's exercising some subtle behavioral differences.